src/configuration.ts

```diff
@@ -317,23 +317,28 @@ function resolveConfigurationPath(filePath: string, relativeTo?: string) {
     }
 }
 
+interface KeyValue {
+    [s: string]: any;
+}
+
 export function extendConfigurationFile(
     targetConfig: IConfigurationFile,
     nextConfigSource: IConfigurationFile,
 ): IConfigurationFile {
-    function combineProperties<T>(targetProperty: T | undefined, nextProperty: T | undefined): T {
+    function combineProperties(
+        targetProperty: KeyValue | undefined,
+        nextProperty: KeyValue | undefined,
+    ): KeyValue {
```

Comment:
I don't think you need a new interface named `KeyValue`... I would prefer keeping the generic type param:

```ts
function combineProperties<T extends { [key: string]: any }>(
    targetProperty: T | undefined,
    nextProperty: T | undefined,
): T {
```

Reply:
```suggestion
    ): T {
```